core/chains/evm/txm/metrics.go

```diff
@@ -95,12 +101,22 @@ func (m *txmMetrics) RecordTimeUntilTxConfirmed(ctx context.Context, duration fl
 	m.timeUntilTxConfirmed.Record(ctx, duration)
 }
 
-func (m *txmMetrics) EmitTxMessage(ctx context.Context, tx common.Hash, fromAddress common.Address, toAddress common.Address, nonce string) error {
+func (m *txmMetrics) EmitTxMessage(ctx context.Context, txHash common.Hash, fromAddress common.Address, tx *types.Transaction) error {
+
+	destAddress, err := m.getDestinationAddress(tx)
```

Comment:
I don't think this is the right approach as anyone, not necessarily SVR, can execute this path. In that case, `Emit` will act as a noop, but we'll see a warning thrown on every transmission. Instead, I would skip the function and write it like this and let the error bubble up, or make it a passthrough if `FwdrDestAddress` is not set:
```suggestion
	meta, err := tx.GetMeta()
	if err != nil {
		return err
	}

	var destAddress string 
	if meta != nil && meta.FwdrDestAddress != nil {
		destAddress = meta.FwdrDestAddress.String()
	}
```

Reply:
Added these guidelines as discussed on Slack. I'll start a discussion with a wider group to see how we should approach Beholder events for something as wide as TXM, this will be a follow-up item/ticket.